File: Forgex_Build/license-tools/FxLicenseCore/Models/LicenseRequestInfo.cs
namespace FxLicenseCore.Models;

/// <summary>
/// 请求授权信息模型。
/// </summary>
/// <remarks>
/// 与后端 LicenseRequestInfo 结构保持一致，
/// 用于现场生成 request-info.json 并提交给授权签发端。
/// </remarks>
public sealed class LicenseRequestInfo
{
    /// <summary>
    /// 产品名称。
    /// </summary>
    public string Product { get; set; } = string.Empty;

    /// <summary>
    /// 版本类型。
    /// </summary>
    public string Edition { get; set; } = string.Empty;

    /// <summary>
    /// 客户码。
    /// </summary>
    public string CustomerCode { get; set; } = string.Empty;

    /// <summary>
    /// 机器码。
    /// </summary>
    public string MachineCode { get; set; } = string.Empty;

    /// <summary>
    /// 操作系统类型。
    /// </summary>
    public string OsType { get; set; } = string.Empty;

    /// <summary>
    /// 主机名。
    /// </summary>
    public string Hostname { get; set; } = string.Empty;

    /// <summary>
    /// 生成时间。
    /// </summary>
    public string GeneratedAt { get; set; } = string.Empty;

    /// <summary>
    /// 工具版本。
    /// </summary>
    public string ToolVersion { get; set; } = string.Empty;
}

File: Forgex_MOM/Forgex_Backend/Forgex_Common_Infra/src/main/java/com/forgex/common/license/LicenseRequestInfo.java
package com.forgex.common.license;

import lombok.Data;

/**
 * 授权请求信息
 * <p>
 * 对应客户端现场生成的 request-info.json 文件结构。
 * </p>
 *
 * @author Forgex Team
 * @version 1.0.0
 */
@Data
public class LicenseRequestInfo {

    /**
     * 产品名称。
     */
    private String product;

    /**
     * 版本类型。
     */
    private String edition;

    /**
     * 客户码。
     */
    private String customerCode;

    /**
     * 机器码。
     */
    private String machineCode;

    /**
     * 操作系统类型。
     */
    private String osType;

    /**
     * 主机名。
     */
    private String hostname;

    /**
     * 生成时间。
     */
    private String generatedAt;

    /**
     * 工具版本。
     */
    private String toolVersion;
}
